Angular/src/app/domains/notification/services/notification-signalr.service.ts

```diff
@@ -54,33 +61,36 @@ export class NotificationSignalRService {
     this.signalRService.addMethod('notification-removeSeveralUnread', (args) => {
       const ids: number[] = JSON.parse(args);
       console.log('%c [Notification] Notification Count', 'color: green; font-weight: bold');
-      ids.forEach( idNum => this.store.dispatch(removeUnreadNotification({ id: idNum })));
+      ids.forEach(idNum => this.store.dispatch(removeUnreadNotification({ id: idNum })));
     });
 
-    this.targetedFeature = {parentKey: this.authService.getCurrentTeamId(TeamTypeId.Site).toString() , featureName : 'notification-domain'};
+    this.targetedFeature = { featureName: 'notification-domain' };
     this.signalRService.joinGroup(this.targetedFeature);
   }
 
   private IsInMyDisplay(notification: Notification) {
     const additionalInfo = this.authService.getAdditionalInfos();
-    const token = this.authService.getUncryptedToken();
-    const okSite: Boolean = notification.siteId === token.userData.currentTeams.find(t => t.teamTypeId == TeamTypeId.Site)?.currentTeamId;
+
     const okUser: Boolean = (notification.notifiedUsers === undefined) ||
-    (notification.notifiedUsers === null) ||
-    (notification.notifiedUsers.length === 0) ||
-    (notification.notifiedUsers.some(u => u.id === additionalInfo.userInfo.id));
-    const okRole: Boolean = (notification.notifiedPermissions === undefined) ||
-    (notification.notifiedPermissions === null) ||
-    (notification.notifiedPermissions.length === 0) ||
-    (notification.notifiedPermissions.some(e => this.authService.hasPermission('' + e.id)));
+      (notification.notifiedUsers === null) ||
+      (notification.notifiedUsers.length === 0) ||
+      (notification.notifiedUsers.some(u => u.id === additionalInfo.userInfo.id));
 
-    return okSite && okUser && okRole;
+    const okRole: Boolean = (notification.notifiedRoles === undefined) ||
+      (notification.notifiedRoles === null) ||
+      (notification.notifiedRoles.length === 0) ||
+      (notification.notifiedRoles.some(e => this.authService.hasPermission('' + e.id)));
```

Comment:
ne pas utiliser this.authService.hasPermission dans Angular/src/app/domains/notification/services/notification-signalr.service.ts
=> il faut verifier les role des teams his.myTeams.some(myTeam => myTeam.id === notifiedTeam.id)